cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/TreeRegexp.java

```diff
@@ -53,7 +53,7 @@
                 nonCapturingMaybe = false;
             } else if (c == '?' && last == '(') {
                 nonCapturingMaybe = true;
-            } else if (c == ':' && nonCapturingMaybe) {
+            } else if ((c == ':' || c == '!') && nonCapturingMaybe) {
```

Comment:
This looks to have fixed this for negatives but not positive lookaheads ('(?=...)'), or look behinds ('(?<=..)' and '(?<!...)').